src/check/parse/tokenize.zig

```diff
@@ -637,6 +640,47 @@ pub const Cursor = struct {
         var sawNewline = false;
         var indent: u16 = 0;
 
+        // SIMD fast path: process chunks of spaces and tabs
+        const SIMD_WIDTH = 16;
+        const Vec16 = @Vector(SIMD_WIDTH, u8);
+
+        while (self.pos + SIMD_WIDTH <= self.buf.len) {
+            const chunk: Vec16 = self.buf[self.pos .. self.pos + SIMD_WIDTH][0..SIMD_WIDTH].*;
+
+            // Use SIMD builtins to check if all characters are either spaces or tabs
+            const spaces: Vec16 = @splat(' ');
+            const tabs: Vec16 = @splat('\t');
+            const is_space = chunk == spaces;
+            const is_tab = chunk == tabs;
+            const true_vec: @Vector(SIMD_WIDTH, bool) = @splat(true);
+            const is_trivial = @select(bool, is_space, true_vec, is_tab);
+            const all_trivial = @reduce(.And, is_trivial);
+
+            // If not all characters are spaces or tabs, fall back to scalar processing
+            if (!all_trivial) {
+                break;
+            }
+
+            // Process the chunk - all characters are spaces or tabs
+            if (sawNewline) {
+                // Use SIMD to count spaces efficiently
+                const ones: @Vector(SIMD_WIDTH, u16) = @splat(1);
+                const zeros: @Vector(SIMD_WIDTH, u16) = @splat(0);
+                const space_count = @reduce(.Add, @select(u16, is_space, ones, zeros));
+                indent += space_count;
+
+                // Handle tabs individually due to tab width alignment complexity
+                for (0..SIMD_WIDTH) |i| {
+                    if (chunk[i] == '\t') {
+                        indent = (indent + self.tab_width) & ~(self.tab_width - 1);
+                    }
+                }
+            }
+            self.pos += SIMD_WIDTH;
+            self.chomp_trivia_simd_chunks_processed += 1;
```

Comment:
I was thinking of putting this behind a debug flag or something so it doesn't impact on release performance at all.